src/routes/users.rs

```diff
@@ -121,6 +122,49 @@ pub async fn mods_list(
     }
 }
 
+#[get("teams")]
+pub async fn teams(
+    req: HttpRequest,
+    info: web::Path<(UserId,)>,
+    pool: web::Data<PgPool>,
+) -> Result<HttpResponse, ApiError> {
+    let id: crate::database::models::UserId = info.into_inner().0.into();
+
+    let current_user = get_user_from_headers(req.headers(), &**pool).await.ok();
+
+    let results;
+    let mut same_user = false;
+
+    if let Some(user) = current_user {
+        if user.id.0 == id.0 as u64 {
+            results =
+                crate::database::models::TeamMember::get_from_user_private(id, &**pool).await?;
```

Comment:
Since this is used a lot, it may be worth having `use crate::database::models::TeamMember;`